pkg/network/render.go

```diff
@@ -689,7 +689,9 @@ func isMigrationChangeSafe(prev, next *operv1.NetworkSpec, infraStatus *bootstra
 	if next.Migration != nil && next.Migration.Mode == operv1.LiveNetworkMigrationMode && infraStatus.HostedControlPlane != nil {
 		return []error{errors.Errorf("live migration is unsupported in a HyperShift environment")}
 	}
-	if prev.Migration != nil && next.Migration != nil && prev.Migration.NetworkType != next.Migration.NetworkType && next.Migration.Mode != operv1.LiveNetworkMigrationMode {
+	if prev.Migration != nil && next.Migration != nil && next.Migration.NetworkType != "" &&
```

Comment:
This now allows changing the NetworkType anytime during the live migration.
Should we at least limit it to going from "set" to "unset"?

Reply:
This patch is actually to allow it to go from "set" to "unset". This happens at the last step of the live migration. When no `migration.featureMigration` is configured, the `migration` field will be removed; therefore, it won't report an error here. However, when the user configures `migration.featureMigration` manually, the `migration` field will not be removed by the server-side apply. Instead, only the `migration.networkType` and `migration.mode` will be unset.